File: python/program/KitchenAppliances.py
from Barang import Barang

class KitchenAppliances(Barang):
    def __init__(self, kode="", nama="", harga="", merk="", garansi="", stok="",
                 DayaListrik="", Berat="", Warna="", Tipe=""):
        super().__init__(kode, nama, harga, merk, garansi, stok)
        self._DayaListrik = DayaListrik
        self._Berat = Berat
        self._Warna = Warna
        self._Tipe = Tipe

    def getDayaListrik(self) -> str:
        return self._DayaListrik
    
    def setDayaListrik(self, DayaListrik: str) -> None:
        self._DayaListrik = str(DayaListrik)

    def getBerat(self) -> str:
        return self._Berat
    
    def setBerat(self, Berat: str) -> None:
        self._Berat = str(Berat)

    def getWarna(self) -> str:
        return self._Warna
    
    def setWarna(self, Warna: str) -> None:
        self._Warna = str(Warna)

    def getTipe(self) -> str:
       return self._Tipe

    def setTipe(self, Tipe: str) -> None:
       self._Tipe = str(Tipe)

    

File: java/program/KitchenAppliances.java
public class KitchenAppliances extends Barang {
    // ini atribut tambahan khusus kitchen appliances
    private String dayaListrik;
    private String berat;
    private String warna;
    private String tipe;

    // constructor, jadi pas bikin objek kita masukin data dari parent + atribut baru
    public KitchenAppliances(String kode, String nama, String harga, String merk, String garansi, String stok,
                             String dayaListrik, String berat, String warna, String tipe) {
        super(kode, nama, harga, merk, garansi, stok); // manggil constructor parent
        this.dayaListrik = dayaListrik;
        this.berat = berat;
        this.warna = warna;
        this.tipe = tipe;
    }

    // getter sama setter biar bisa akses dan ubah data
    public String getDayaListrik() {
      return this.dayaListrik;
   }

   public void setDayaListrik(String dayaListrik) {
      this.dayaListrik = dayaListrik;
   }

   public String getBerat() {
      return this.berat;
   }

   public void setBerat(String berat) {
      this.berat = berat;
   }

   public String getWarna() {
      return this.warna;
   }

   public void setWarna(String warna) {
      this.warna = warna;
   }

   public String getTipe() {
      return this.tipe;
   }

   public void setTipe(String tipe) {
      this.tipe = tipe;
   }
}
